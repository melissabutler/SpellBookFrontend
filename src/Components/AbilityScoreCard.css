.AbilityScoreCard {
    width: 150px;
    height: 130px;
    background-color: blue;
    border-radius: 20px;
    place-content: center;
    display: inline-block;
    margin: 10px;
}