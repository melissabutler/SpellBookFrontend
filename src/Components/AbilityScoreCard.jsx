import React from "react";

import "./AbilityScoreCard.css"


import { Container, Row, Col } from 'react-bootstrap';

const AbilityScoreCard = ({ability, score, mod}) => {
    return (
        <Container className="AbilityScoreCard">
        <Row >
            <h4 className="AbilityScoreCard-ability">{ability}</h4>
        </Row>
        <Row>
            <Col></Col>
            <Col xs={7}>
            <h5 className="AbilityScoreCard-score">{score}</h5>
            </Col>
            <Col></Col>
            
        </Row>
        <Row>
            <Col></Col>
            <Col>
            <h6 className="AbilityScoreCard-mod">{mod}</h6>
            </Col>
            <Col></Col>
            
        </Row>

        </Container>
        
    )
}

export default AbilityScoreCard;