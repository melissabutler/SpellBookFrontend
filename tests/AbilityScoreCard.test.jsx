import React, { useContext } from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import AbilityScoreCard from "../src/Components/AbilityScoreCard";

describe("Ability Score Card", function() {


    it("renders without crashing", async function() {
        render(<AbilityScoreCard />)
    })

    it("matches snapshot", function() {
        const { asFragment } = render(<AbilityScoreCard />);
        expect(asFragment()).toMatchSnapshot();
    })

    it("displays supplied information", function() {
        render(<AbilityScoreCard ability="charisma" score="10" mod="+2"/>)
        
        expect(screen.getByText('charisma')).toBeInTheDocument();
        expect(screen.getByText('10')).toBeInTheDocument();
        expect(screen.getByText('+2')).toBeInTheDocument();
    })



})
