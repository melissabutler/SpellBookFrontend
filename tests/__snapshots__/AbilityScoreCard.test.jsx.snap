// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ability Score Card > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="AbilityScoreCard container"
  >
    <div
      class="row"
    >
      <h4
        class="AbilityScoreCard-ability"
      />
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="col-7"
      >
        <h5
          class="AbilityScoreCard-score"
        />
      </div>
      <div
        class="col"
      />
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="col"
      >
        <h6
          class="AbilityScoreCard-mod"
        />
      </div>
      <div
        class="col"
      />
    </div>
  </div>
</DocumentFragment>
`;
